src/main/components/pages/Withdraw.js

```diff
@@ -1,10 +1,360 @@
+      if(address.match(/[45][0-9AB][123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]{93}/)){
+        setEnteredWithdrawAddressIsValid(true);
+      } else {
+        setEnteredWithdrawAddressIsValid(false);
+      }
+      setEnteredWithdrawAddress(address);
+      setEnteredWithdrawAddressText(address);
+
+  }
+  
+  const changeWithdrawAmount = function(amount) {
+    
+    //Re-add checking for invalid values (non-numbers, <1AU or >availBal, etc
+    setEnteredWithdrawAmount(amount);
+    setEnteredWithdrawAmountText(amount);
+
+  }
+  
+  const changeWithdrawAmountWithText = function(amount, text) {
+    setEnteredWithdrawAmount(amount),
+    setEnteredWithdrawAmountText(text)
+
+  }
+  
+  // Runs when the user clicks "Send all" above the withdraw send amount field
+  const prepareToSendAllFunds = function() {
+    changeWithdrawAmountWithText(props.availableBalance, withdrawAmountSendAllText);
+    setUsingAllFunds(true);
+  }
+  
+  const createWithdrawTx = async function() {
+    
+    // someFunctionToSetWithdrawPageButtonToSpinnyWheel()
+    
+    let withdraw = null;
+    
+    setWithdrawTxStatus("creating");
+    let txCreationWasSuccessful = true;
+    
+    try {
+      if(usingAllFunds) {
+        withdrawTx = await props.wallet.sweepUnlocked({
+          address: enteredWithdrawAddress
+        });
+      } else {
+	
+	/* withdrawTx will always be an array of length 1 if the user did not press the "send all" button
+	 * However, use an array just the same for consistency with a full balance withdraw
+	 * as sweepUnlocked returns an array of Txs
+	 * 
+	 * Later code will then need not distinguish between a withdrawTx created by createTx vs sweepUnlocked
+	 */
+	withdrawTx = new Array(1);
+	withdrawTx[0] = await props.wallet.createTx({
+	  address: enteredWithdrawAddress,
+	  amount: enteredWithdrawAmount,
+	  accountIndex: 0
+	});
+
+      }
+    } catch(e) {
+      console.log("Error creating tx: " + e);
+      setWithdrawTxStatus("");
+      txCreationWasSuccessful = false;  
+    }
+    
+    if(txCreationWasSuccessful){
+      
+      let newWithdrawInfo = new Array(withdrawTx.length);
+      for(let i = 0; i < newWithdrawInfo.length; i++){
+        newWithdrawInfo[i] = {
+          withdrawAddress: enteredWithdrawAddress,
+          withdrawAmount: MoneroUtils.atomicUnitsToXmr(withdrawTx[i].getOutgoingAmount()),
+          withdrawFee: MoneroUtils.atomicUnitsToXmr(BigInteger(withdrawTx[i].getFee().toString())),
+          withdrawHash: withdrawTx[i].getHash(),
+          withdrawKey: withdrawTx[i].getKey()
+        };
+      }
+      setWithdrawInfo(newWithdrawInfo);
+      setWithdrawTxStatus("");
+    }
+  }
+  
+  const relayWithdrawTx = async function() {
+    setWithdrawTxStatus("relaying");
+    
+    let relayTxWasSuccessful = true;
+    let newWithdrawInfo = [...withdrawInfo];
+    
+    for (let i = 0; i < newWithdrawInfo.length; i++){
+      await props.wallet.relayTx(withdrawTx[i]).catch (
```

Comment:
withdrawTx is null here, and exception invoking await should be handled in try...catch block.